corehq/apps/registration/static/registration/js/user_login_form.js

```diff
@@ -118,8 +120,12 @@ hqDefine('registration/js/user_login_form', [
             self.proceedToNextStep();
         };
 
-        self.continueToSsoLogin = function (sso_url) {
-            window.location = sso_url;
+        self.continueToSsoLogin = function (ssoUrl) {
+            if (self.nextUrl) {
+                // note ssoUrl already contains ?username=foo
+                ssoUrl = ssoUrl + "&next=" + self.nextUrl;
```

Comment:
Is `self.nextUrl` already URL-encoded?  If so, this is fine. Otherwise, I think this could be a URL injection concern.

Reply:
I believe it is encoded. here is where `urlParams` is initialized.
https://github.com/dimagi/commcare-hq/blob/9fbba6c41884b79174e93c66e9254b0566940dc5/corehq/apps/registration/static/registration/js/login.js#L24

Reply:
I see, looks like it's not ~encoded~ quoted (sorry, meant "quoted" originally).

Reply:
it does not need to be quoted. see comment below.